src/PrestaShopBundle/Controller/Admin/SpecificPriceController.php

```diff
@@ -96,6 +101,78 @@ public function addAction(Request $request)
         return $response;
     }
 
+    /**
+     * Get one specific price list for a product
+     *
+     * @Template("@PrestaShop/Admin/Product/ProductPage/Forms/form_specific_price.html.twig")
+     *
+     * @param int $idSpecificPrice
+     *
+     * @return array
+     */
+    public function getUpdateFormAction($idSpecificPrice)
+    {
+        /** @var AdminProductWrapper $adminProductWrapper */
+        $adminProductWrapper = $this->get('prestashop.adapter.admin.wrapper.product');
+        try {
+            $price = $adminProductWrapper->getSpecificPriceDataById($idSpecificPrice);
+        } catch (\PrestaShopObjectNotFoundException $e) {
```

Comment:
use namespace, will be easier to refactor.
More don't use Legacy exception in Symfony Context.